receiver/k8sclusterreceiver/metadata.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclusterreceiver
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Keys for K8s properties
+	k8sKeyWorkLoad     = "k8s.workload"
+	k8sKeyWorkLoadName = "k8s.workload.name"
+)
+
+// KubernetesMetadata currently provides a reasonable way to store metadata
+// and may changes depending on how we decide to handle property updates.
+type KubernetesMetadata struct {
+	resourceIDKey string
+	resourceID    string
+	properties    map[string]string
+}
+
+// getGenericMetadata is responsible for collecting metadata from K8s resources that
+// live on v1.ObjectMeta.
+func getGenericMetadata(om *v1.ObjectMeta, resourceType string) *KubernetesMetadata {
+	properties := om.Labels
```

Comment:
Is it OK that we modify `om.Labels` map in this function? Do the caller expect that? Maybe add a comment that warns about since it is not apparent from the signature.

Reply:
A new map should have been created inside the method and that map should be returned. Updated.